nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/hazelcast/HazelcastCore.java

```diff
@@ -513,13 +527,15 @@ private void buildNetworkConfiguration(Config config, boolean hostAwarePartition
         } else {
             //build the domain discovery config
             config.setProperty("hazelcast.discovery.enabled", "true");
-            PartitionGroupConfig partitionGroupConfig = config.getPartitionGroupConfig();
-            partitionGroupConfig.setEnabled(true);
-            partitionGroupConfig.setGroupType(PartitionGroupConfig.MemberGroupType.SPI);
             config.getNetworkConfig().getJoin().getDiscoveryConfig().addDiscoveryStrategyConfig(
                     new DiscoveryStrategyConfig(DomainDiscoveryStrategy.class.getName())
                             .addProperty(HOST_AWARE_PARTITIONING.key(), hostAwarePartitioning));
             config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
+            if (Boolean.parseBoolean(System.getProperty("hazelcast.auto-partition-group", "false"))) {
+                PartitionGroupConfig partitionGroupConfig = config.getPartitionGroupConfig();
+                partitionGroupConfig.setEnabled(true);
+                partitionGroupConfig.setGroupType(PartitionGroupConfig.MemberGroupType.SPI);
+            }
```

Comment:
Isn't this changing the default behaviour even with CP subsytem turned off?
Now the partitions will default to "member" with each instance being in its own partition group instead of being determined by the DomainDiscoveryStrategy which utilises the instance group attribute and is optionally host-aware.

Without this it won't be host aware and the instance groups will be ignored for partition purposes - I'm not convinced this is an improvement.